redisinsight/ui/src/pages/browser/components/stream-details/StreamDetails/StreamDetails.tsx

```diff
@@ -43,26 +44,38 @@ const StreamDetails = (props: Props) => {
   const { data: entries = [], columns = [], onClosePopover, isFooterOpen } = props
   const dispatch = useDispatch()
 
-  const { loading } = useSelector(streamSelector)
+  const { loading, start, end } = useSelector(streamSelector)
   const {
     total,
     firstEntry,
     lastEntry,
   } = useSelector(streamDataSelector)
   const { name: key } = useSelector(selectedKeyDataSelector) ?? { name: '' }
 
-  const [sortedColumnName, setSortedColumnName] = useState('id')
-  const [sortedColumnOrder, setSortedColumnOrder] = useState(SortOrder.DESC)
+  const [sortedColumnName, setSortedColumnName] = useState<string>('id')
+  const [sortedColumnOrder, setSortedColumnOrder] = useState<SortOrder>(SortOrder.DESC)
 
   const loadMoreItems = () => {
-    const lastLoadedEntryId = last(entries)?.id ?? ''
-    const lastEntryId = sortedColumnOrder === SortOrder.ASC ? lastEntry?.id : firstEntry?.id
+    const lastLoadedEntryId = last(entries)?.id
+    const lastLoadedEntryTimeStamp = getTimestampFromId(lastLoadedEntryId)
+
+    const lastRangeEntryTimestamp = end ? parseInt(end, 10) : getTimestampFromId(lastEntry?.id)
+    const firstRangeEntryTimestamp = start ? parseInt(start, 10) : getTimestampFromId(firstEntry?.id)
+    const shouldLoadMore = () => {
+      if (!lastLoadedEntryTimeStamp) {
+        return false
+      }
+      return sortedColumnOrder === SortOrder.ASC
+        ? lastLoadedEntryTimeStamp > lastRangeEntryTimestamp
+        : lastLoadedEntryTimeStamp < firstRangeEntryTimestamp
+    }
 
-    if (lastLoadedEntryId && lastLoadedEntryId !== lastEntryId) {
+    if (shouldLoadMore()) {
       dispatch(
         fetchMoreStreamEntries(
           key,
-          `${xrangeIdPrefix + lastLoadedEntryId}`,
+          sortedColumnOrder === SortOrder.DESC ? start : `${xrangeIdPrefix + lastLoadedEntryId}`,
+          sortedColumnOrder === SortOrder.DESC ? `${xrangeIdPrefix + lastLoadedEntryId}` : end,
```

Comment:
could you please create variable for
``${xrangeIdPrefix + lastLoadedEntryId}``
also you can pass these values as 1 parameter ([] or {})